### PR TITLE
Expand backend equivalence testing

### PR DESCRIPTION
**Pull request by FelipeNuti**
# Checklist

- [ ] Added a `CHANGELOG.rst` entry

### CODE REVIEW

**Comment by FelipeNuti** [comment]
Currently the tests are failing due to the absence of a BigQuery key file `bq_key.json`, which we used to test locally but obviously couldn't push to the repo. For now the best course of action is likely to just leave the BigQuery tests out, and then figure out how to test this component of the backend equivalence through GitHub actions.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -32,11 +27,33 @@ jobs:
...
+          - 5432:5432
     strategy:
       matrix:
         os: [Ubuntu]
         python-version: ['3.8', '3.9', '3.10']
     steps:
+      - name: Docker login
+        run: docker login ghcr.io -u "${{ github.actor }}" -p "${{ secrets.GITHUB_TOKEN }}"
+      - name: Pull image
+        run: docker pull ghcr.io/quantco/qc-mambaforge-base:latest
```
Comment (windiana42) [inline .github/workflows/ci.yml:56]: we should get rid of this at some point again

File: pyproject.toml [hunk 2]
```diff
@@ -20,17 +20,23 @@ classifiers = [
 ]
 
 [tool.poetry.dependencies]
-python = "^3.8"
+python = ">=3.8,<3.11"
```
Comment (windiana42) [inline pyproject.toml:23]: this requires more explicit github actions with different python versions

File: tests/test_backend_equivalence.py [hunk 3]
```diff
@@ -72,34 +73,80 @@ def pandas_impls():
...
+
+# def bigquery_impls():
+#     dataset = "dataset_test"
+#     project_id = "project-id"
+#     auth_path = "bq_key.json"
+#     credentials = service_account.Credentials.from_service_account_file(auth_path)
+#     local_conn = (  # ?DataSetId={dataset}&ProjectId={project_id}&InitiateOAuth=GETANDREFRESH&OAuthSettingsLocation={auth_path}"
+#         f"bigquery://{project_id}"
+#     )
+#     return sql_conn_to_impls(local_conn, project_id=project_id, dataset=dataset)
```
Comment (windiana42) [inline tests/test_backend_equivalence.py:122]: let's keep this code commented here until we decided how to setup cloud databases in github actions

**Review by windiana42 — APPROVED** [review]
Approved for pulling into separate branch 

This branch will be used for making it work with main branch.

File: pyproject.toml [hunk 4]
```diff
@@ -20,17 +20,23 @@ classifiers = [
...
 numpy = "^1.23.1"
 pandas = "^1.4.3"
-SQLAlchemy = "^1.4.39"
+SQLAlchemy = "^1.4.27"
+mutate = "^0.0.1-alpha.13"
+google-auth = "^2.9.1"
```
Comment (NMAC427) [inline pyproject.toml:28]: Dependencies for different backhands should not be mandatory. Instead they should be installed using extras (eg `pip install pydiverse.pipedag[bigquery]`

File: pyproject.toml [hunk 5]
```diff
@@ -20,17 +20,23 @@ classifiers = [
...
 [tool.poetry.dev-dependencies]
 pytest = "^7.1.2"
+#pyodbc = "^4.0.34"
 pytest-xdist = "^2.5.0"
+psycopg2 = "2.9.3"
 black = "^22.6.0"
 pre-commit = "^2.20.0"
 isort = "^5.10.1"
+sqlalchemy-bigquery = "^1.4.4"
+google-auth = "^2.9.1"
```
Comment (NMAC427) [inline pyproject.toml:39]: Duplicate dependencies

Comment (windiana42) [reply pyproject.toml:39]: I will just push to separate branch and then we can open a new PR from there to get it on main.

**Pull request merged**